rclcpp/include/rclcpp/node_interfaces/node_graph_interface.hpp

```diff
@@ -25,10 +26,45 @@
 
 #include "rclcpp/event.hpp"
 #include "rclcpp/macros.hpp"
+#include "rclcpp/qos.hpp"
 #include "rclcpp/visibility_control.hpp"
+#include "rmw/topic_endpoint_info.h"
 
 namespace rclcpp
 {
+
+enum class EndpointType
+{
+  Invalid = RMW_ENDPOINT_INVALID,
+  Publisher = RMW_ENDPOINT_PUBLISHER,
+  Subscription = RMW_ENDPOINT_SUBSCRIPTION
+};
+
+/**
+ * Use to get topic endpoint information that containing the node name, node namespace, topic type,
```

Comment:
nitpick: this is hard to read, and I'm not sure what it's trying to say

"Struct that contains topic endpoint information like the associated node name, node namespace, ..., and its QoS."?

Reply:
Done.